docs/deploy.rst

```diff
@@ -1,7 +1,7 @@
 .. _deploy:
 
-Deploy
-======
+Deployment Guide
+================
```

Comment:
I know it wasn't added as a part of this PR, but there is a lot of detail about Google Container Engine.  Do we want that level of detail here, and do we want to support all these details for the future?  When is the last time anyone tried this?

Reply:
agreed.

I don't want to drop the instructions on EB because we are actively using that (though maybe this thinking doesn't scale, because everyone could have their set of favored deployments...), but I'd be fine splitting those out somewhere else.  If this specific page could have just the most important or canonical deployment types, that would make it easier to read through.

For the GCE stuff specifically, I say we remove it.

Reply:
My (perhaps unpopular) opinion on this is that the detailed instructions for a given deployment platform could be made into a tutorial that's targeted to "Deploying Girder on ElasticBeanstalk" or whatever. Having a multitude of those is not a problem (or at least, not as much of a problem as stuffing all of them into the `Deployment Guide` section).

And keep in mind, my use of the word "tutorial" is pretty fluid - when I say "tutorial", the thing I'm thinking of might be called a "recipe" or "developer cookbook entry" by someone else.

Reply:
Each tutorial would have the following metadata:
* known to work with Gider X.Y.Z
* last updated on YYYY-MM-DD
* last tested on YYYY-MM-DD
* authors: John Doe (Foo Inc), ...

Reply:
Honestly, that's the very thing that worries me about the idea of having a whole bunch of tutorials with little curation -- for a newcomer developer, they are just presented with some large, overwhelming list of links, and then they have to try to sort through which ones are actually up to date. I'd rather have a smaller list where everything is known (or at least expected) to work with the version that the docs correspond to. We have a limited team of maintainers, and doc maintenance is not something we historically have a ton of resources to allocate toward.

Reply:
> I'd rather have a smaller list where everything is known (or at least expected) to work with the version that the docs correspond to.

Of course.

> doc maintenance is not something we historically have a ton of resources to allocate toward.

Which is sad ... without explaining and documenting how to use a (open source) project, its adoption and use internal or external will fade ... 

The good news is that Girder has been a great example regarding documentation

I like to say *If it is not tested and documented, it is broken*



All of that said, I was just sharing my experience acquired while tackling similar issue with 3D Slicer. 

It is still nice to keep track of half backed tutorials, experiments, etc .. these simply need to be linked of a place labeled as such and not prominently visible from the documentation landing page. As you told, we do not want to confuse or overwhelm new comers. 

To help move the link around, understanding when an experiment/tutorial/etc .. has been created, last tested is quite important. With the resources at hand, we can't always manually test everything or write test for it. 

I think that this rather easy to add metadata will help make inform (and unbiased) decisions. 

Idea: These "half baked" experiments/tutorials/... could be linked of a "Girder Labs" Github wiki page ?

Reply:
I'm actually a huge fan of the external 'tutorial' idea for more exotic deployment scenarios. This documentation should cover whatever we consider to be the 'canonical' deployment with a section that essentially advertises that other deployments are possible (with links to tutorials).  

This also provides an opportunity to say something like "If you're interested in more complex deployments please send us an email,  or drop by our gitter channel blah blah blah"  I want to be *very* careful to not come across as doing a direct sell,  but i don't think its bad to communicate "We've done this in the past and we'd be happy to help you do it now"